integration_tests/test_nft_transfer.py

```diff
@@ -0,0 +1,211 @@
+            "transfer",
+            "nft",
+            src_channel,
+            addr_dst,
+            denomid,
+            tokenid,
+            "-y",
+            home=cli_src.data_dir,
+            from_=addr_src,
+            keyring_backend="test",
+            chain_id=cli_src.chain_id,
+            node=cli_src.node_rpc,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    # FIXME more stable way to wait for relaying
+    time.sleep(20)
+
+    # get class hash on destination chain
+    class_hash = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft-transfer",
+            "class-hash",
+            "nft/" + dst_channel + "/" + denomid,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )["hash"]
+
+    dst_denom_id = "ibc/" + class_hash
+
+    # query denom on destination chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft",
+            "denom",
+            dst_denom_id,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == denomuri, rsp["uri"]
+
+    # query nft on destination chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft",
+            "token",
+            "ibc/" + class_hash,
+            tokenid,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == tokenuri, rsp
+    assert rsp["owner"] == addr_dst, rsp
+
+    # transfer nft back to source chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "tx",
+            "nft-transfer",
+            "transfer",
+            "nft",
+            dst_channel,
+            addr_src,
+            dst_denom_id,
+            tokenid,
+            "-y",
+            home=cli_dst.data_dir,
+            from_=addr_dst,
+            keyring_backend="test",
+            chain_id=cli_dst.chain_id,
+            node=cli_dst.node_rpc,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    # FIXME more stable way to wait for relaying
+    time.sleep(20)
+
+    # nft should be burnt on destination chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft",
+            "collection",
+            dst_denom_id,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )["collection"]
+
+    assert len(rsp["nfts"]) == 0, rsp
+
+    # query nft on source chain
+    rsp = json.loads(
+        cli_src.raw(
+            "query",
+            "nft",
+            "token",
+            denomid,
+            tokenid,
+            home=cli_src.data_dir,
+            node=cli_src.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == tokenuri, rsp
+    assert rsp["owner"] == addr_src, rsp
```

Comment:
adding some tests which ensure bad arguments not working.
such as
very long length tokenid, denomid?
some invalid data